WIP: If /boot/firmware is in fstab, use it as BOOTFS

XECDesign opened the pull request:
Debian's raspi-firmware package assumes the boot partition is mounted as /boot/firmware/ rather than /boot/

This commit adds support for either approach by default

However, might not be the most sensible implementation...

timg236 commented on rpi-eeprom-update-default:
```diff
@@ -3,6 +3,11 @@ FIRMWARE_ROOT=/lib/firmware/raspberrypi/bootloader
-BOOTFS=/boot
 USE_FLASHROM=0
 EEPROM_CONFIG_HOOK=
+
```
Would it be better to just remove BOOTFS from rpi-eeprom-update-default then do then for loop in the script if BOOTFS is not defined?
That would still allow people to supply a default value.

XECDesign commented on rpi-eeprom-update:
```diff
@@ -576,6 +575,12 @@ findBootFS()
       TMP_BOOTFS_MNT="$(mktemp -d)"
       mount /dev/mmcblk0p1 "${TMP_BOOTFS_MNT}"
       BOOTFS="${TMP_BOOTFS_MNT}"
+   elif [ -z ${BOOTFS+set} ]; then
```
"parameter expansion" - https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html

`${parameter+word}` becomes `word` if `$parameter` is set, but otherwise returns an empty string. It helps draw a distinction between an empty variable and an unset variable.

Supporting `BOOTFS=` makes no sense, so we could make it more readable by removing the use of a slightly obscure feature, if that's preferred.

timg236 replied:
ok

By "Supporting BOOTFS= makes no sense" are you saying that it shouldn't be possible to set a hardcoded path via the defaults file (which seems useful in buildroot) or just that the previous BOOTFS default value made no sense?

XECDesign replied:
I mean that you wouldn't set BOOTFS to an empty string and expect things to work. So `-z "${BOOTFS}"`  would work just as well as `-z ${BOOTFS+set}` and would be easier to understand at a glance.

timg236 replied:
ok 👍

XECDesign commented:
I've removed the `+set` to keep things simple and added support for a new script to abstract away the boot location in one spot rather than have it hard-coded all over the place ( see https://github.com/RPi-Distro/raspberrypi-sys-mods/issues/74). That script isn't in any public release of sys-mods yet, but will be in bookworm.

timg236 commented:
Sounds ok, remove WIP when you are happy to merge. We can old this in the github repo for a bit in order to give other distro maintainers a chance to comment

timg236 commented:
@XECDesign - Did you want me to merge this?

XECDesign commented:
I've been been running with the change for a little while and haven't seen any issues
